docs/doc_sources/reference_guides/dpctl/program.rst

```diff
@@ -0,0 +1,37 @@
+.. _dpctl_program_pyapi:
+
+:py:mod:`dpctl.program`
+=======================
+
+:py:mod:`dpctl.program` provides a way to create a SYCL kernel
+from either an OpenCL program source code represented as a string
+or a SPIR-V binary file.
+
+It implements creation of interoperability
+``sycl::kernel_bundle<sycl::bundle_state_executable>`` (a collection of kernels),
+as well as creation of individual ``sycl::kernel``, suitable for submission for
+execution via :py:meth:`dpctl.SyclQueue.submit`.
+
+.. py:module:: dpctl.program
+
+.. currentmodule:: dpctl.program
+
+.. autosummary::
+    :toctree: generated
+    :nosignatures:
+
+    create_program_from_source
+    create_program_from_spirv
```

Comment:
Generated page for create_program_from_spirv lists parameters incorrectly
![Screenshot 2024-04-02 110019](https://github.com/IntelPython/dpctl/assets/46709016/2ceb75c7-13c6-45bc-afbe-fa9d5f90ba8a)
should be IL[const char]

Reply:
Docstring of `create_program_from_spirv` needs to be adjusted. It specifies C type for the argument, but Python type is expected. 

I will push this to docstring-only branch and then will rebased this branch